channeldb/channel.go

```diff
@@ -538,6 +548,16 @@ type OpenChannel struct {
 	// for which we are the initiator.
 	FundingTxn *wire.MsgTx
 
+	// LocalShutdownScript is set to a pre-set script if the channel was opened/
+	// by the local node with option_upfront_shutdown_script set. If the option/
+	// was not set, the field is empty.
+	LocalShutdownScript lnwire.DeliveryAddress
```

Comment:
What do you think about making the methods `LocalShutdownScript()` fetch directly from the database instead of this pre-populated field? Would it have any downsides?

Reply:
The fields in `OpenChannel` are also used for writing to the DB, and threading the values through the funding flow (as part of chanState). Since we need the fields in `OpenChannel` to write it makes sense to read them in/accessin memory?

To keep them out of channel state completely, they'd need to be in `ChannelReservation` which leaks into `lnwallet`.